api/v1/tetragon/tetragon.proto

```diff
@@ -259,6 +259,8 @@ message Process {
     ProcessCredentials process_credentials = 17;
     // Executed binary properties. This field is only available on ProcessExec events.
     BinaryProperties binary_properties = 18;
+    // User name associated with process.
```

Comment:
After some thoughts and let's see what @jrfastab thinks too, but since username is something abstracted from a user space (name) services and not kernel related that could do network lookups in future, let's separate the things.

I suggest something like this:
```
// User records
message UserRecord {
   // The UNIX username for this record. Corresponds to `pw_name` field of [struct passwd](https://man7.org/linux/man-pages/man3/getpwnam.3.html)
   // and the `sp_namp` field of [struct spwd](https://man7.org/linux/man-pages/man3/getspnam.3.html).
   name = 1;
}
 
// Unix user account information
message UserAccount {
   // UserRecord user = 1;
}
```

Then inside message Process:
```
   // Unix user account information. Available only on host deployments when Tetragon is running as a system service or directly on host.
   UserAccount account = 19;
```

@jrfastab I don't like the 'account' name field, feel free to suggest.

All this is to separate things as username is something that is available depending on the workflow but also to be extensible in future with where things are going in the userspace stack:
* https://systemd.io/USER_GROUP_API/
* https://systemd.io/USER_RECORD/

Also eventually there are some fields there that we could add to records so we offer cool features to our users ;-)

Reply:
Thank you for such detailed review! Your proposed solution with `UserAccont` struct looks great. It allows us to extend event with some other useful information in future.  I could't find a better naming for 'Account'. So I keep this name for now.

Reply:
What do we get by having  this couple layers of structures here?

 UserRecord {
        UserName {
               name string
        }
  }
  
  I get having one struct any reason to not just embed name in UserRecord?
  
   UserRecord {
      name string
    }

If user record grows fine.

Reply:
Yeh having such layering may seem too much, but in the other hand the abstraction at userspace has always been separate users and groups, you have two files for /etc/passwd and /etc/groups, at systemd layer two separate records, glibc [sgrp](https://elixir.bootlin.com/glibc/glibc-2.38/source/gshadow/gshadow.h#L36) and spwd, NSS etc... and I'm trying to position tetragon be autocompatible as possible... ;-) 

Since you point this, I did another review of those user space fields, systemd is the most advanced one. We won't do NSS lookups in tetragon. So taking current distros support:
- Arch Linux seems to have already adopted [systemd-userdb service](https://man.archlinux.org/man/core/systemd/systemd-userdbd.8.en) JSON User/Group Record Query Multiplexer/NSS Compatibility as part of the systemd core.
- Ubuntu has https://manpages.ubuntu.com/manpages/lunar/man8/systemd-userdbd.8.html but it is provided by another package. Anyway the end result is a socket on  `/run/systemd/userdb/` that provides json records.
- Debian issue closed by packaging systemd-userdb in its own package https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=976960

So I see two choices here, and obviously I go with json systemd compatible one as it will allow us to support NSS and all new stuff https://systemd.io/USER_RECORD/ and https://systemd.io/GROUP_RECORD/ by querying the socket automatically, then mapping fields to the tetragon fields automatically. 

1. JSON systemd-userdb compatible two structs: per user and one per group:
```
// User records
UserRecord {
   name string
   ...
}

// Group records
GroupRecord {
   ...
}

message Process {
   // Unix user account information. Available only on host deployments when Tetragon is running as
   // a system service or directly on host. Accounts information is retrieved from traditional user database
   // `/etc/passwd` and no name services lookups are performed.
   UserRecord user = 19;

   // Unix group account information....
   GroupRecord group = xx;
}
```

2. Or we do simple one struct and embed everything, and prefix fields with either user/group for ones that have same name and also same meaning we let it grow
```
// Unix user account information
AccountRecord {
    username string // The UNIX user name for this record.
    groupname string // The UNIX group name for this record.
}

message Process {
   // Unix user account information. Available only on host deployments when Tetragon is running as
   // a system service or directly on host. Accounts information is retrieved from traditional user database
   // `/etc/passwd` and no name services lookups are performed.
   AccountRecord user/account = 19
}
```

Again for obvious reasons, I prefer 1. WDYT @jrfastab @anfedotoff ?

Reply:
I vote for the first one:). Btw, if we add group record, I think I can fill it the same way as user record.

Reply:
> I vote for the first one:). Btw, if we add group record, I think I can fill it the same way as user record.

Yes same way, just mention it in the code doc.

Let's wait for @jrfastab ack and I hope we have sorted out all stuff ;-)

Reply:
1 works for me. For simplicity we could start with just the User record in t his PR and follow up with Group?

Reply:
> 1 works for me. For simplicity we could start with just the User record in t his PR and follow up with Group?
It's no problem to get primary group name for the user using [LookupGroupId](https://pkg.go.dev/os/user#LookupGroupId). But primary group name is often the same as user name. So I not sure if it is useful to have only this info.

Reply:
So, vote to keep only User record for now, but for me it is OK to add Group record as I described. @tixxdz, what do you think?

Reply:
@anfedotoff Up to you if you can make it shorter, document it and covered with tests then go with it, otherwise we can merge this PR first, then you follow up.